### DISCUSSION TITLE
What is the deal with sessions?

### DISCUSSION THREAD

**Discussion by ghost** [nksaraf/vinxi#416] [Q&A] [answered]
Currently looking to build a web app with SolidStart, but I cant find anything descriptive about how Vinxi actually handles session state. Just a few abstractions and one sentence about each.

Is there any way to actually configure, say, a SQL table or redis cache to hold sessions? Im finding it odd that there seems to be zero options to extend or adjust this while also having no explanation of what the current implementation does.

**Comment by ghost** [comment]
Okay so it looks like vinxi just repackages h3 sessions. for anyone interested:

https://h3.unjs.io/examples/handle-session

its extremely limited. im not familiar with the package so i could be missing something but it appears that it just stores session data in memory, and disappears when the app terminates. really wish this was clarified more in the docs.

**Reply by datner** [reply, marked as answer]
This is incorrect, it stores the session data on a cookie. Which is an issue since that means you're even more limited than before, since you need your data to be serializable _and_ be encoded in less than 4096 chars

**Reply by ghost** [reply]
Really weird. Wish there were better explanations of this. Calling what is essentially a JWT a "session" seems a bit off.
